apps/server/test/lexical/server/code_intelligence/entity_test.exs

```diff
@@ -299,30 +304,217 @@ defmodule Lexical.Server.CodeIntelligence.DefinitionTest do
+    test "excludes the module segments after the hovered separator", %{project: project} do
+      code = ~q[
+        AAA|.BBB.CCC.DDD.EEE
+        ]
+
+      assert {:ok, {:module, AAA}} = resolve(project, code)
+
+      code = ~q[
+        AAA.BBB|.CCC.DDD.EEE
+        ]
+
+      assert {:ok, {:module, AAA.BBB}} = resolve(project, code)
+
+      code = ~q[
+        AAA.BBB.CCC|.DDD.EEE
+        ]
+
+      assert {:ok, {:module, AAA.BBB.CCC}} = resolve(project, code)
+
+      code = ~q[
+        AAA.BBB.CCC.DDD|.EEE
+        ]
+
+      assert {:ok, {:module, AAA.BBB.CCC.DDD}} = resolve(project, code)
+    end
+
+    test "returns module segments in nodes spanning multiple lines", %{project: project} do
+      code = ~q[
+        foo =
+          On.Another.Lin|e
+        ]
+
+      assert {:ok, {:module, On.Another.Line}} = resolve(project, code)
+
+      code = ~q[
+        On|.
+          Multiple.
+          Lines
+        ]
+
+      assert {:ok, {:module, On}} = resolve(project, code)
+
+      code = ~q[
+        On.
+          Multiple.
+          Line|s
+        ]
+
+      assert {:ok, {:module, On.Multiple.Lines}} = resolve(project, code)
+
+      code = ~q[
+        Enu|m.map(1..10, fn i ->
+          i + 1
+        end)
+        ]
+
+      assert {:ok, {:module, Enum}} = resolve(project, code)
+    end
+
+    test "resolves explicit aliases", %{project: project} do
+      code = ~q[
+        defmodule Example do
+          alias Long.Aliased.Module
+          Modul|e
+        end
+        ]
+
+      assert {:ok, {:module, Long.Aliased.Module}} = resolve(project, code)
+
+      code = ~q[
+        defmodule Example do
+          Modul|e
+          alias Long.Aliased.Module
+        end
+        ]
+
+      assert {:ok, {:module, Module}} = resolve(project, code)
+    end
+
+    test "resolves implicit aliases", %{project: project} do
+      code = ~q[
+        defmodule Example do
+          defmodule Inner do
+          end
+
+          Inne|r
+        end
+        ]
+
+      assert {:ok, {:module, Example.Inner}} = resolve(project, code)
+
+      code = ~q[
+        defmodule Example do
+          Inne|r
+
+          defmodule Inner do
+          end
+        end
+        ]
+
+      assert {:ok, {:module, Inner}} = resolve(project, code)
+
+      code = ~q[
+        defmodule Example do
+          __MODULE_|_
+        end
+        ]
+
+      assert {:ok, {:module, Example}} = resolve(project, code)
+
+      code = ~q[
+        defmodule Example do
+          |__MODULE__
+        end
+        ]
+
+      assert {:ok, {:module, Example}} = resolve(project, code)
+
+      code = ~q[
+        defmodule Example do
+          __MODULE__.Neste|d
+        end
+        ]
+
+      assert {:ok, {:module, Example.Nested}} = resolve(project, code)
+
+      code = ~q[
+        defmodule Example do
+          __MODULE_|_.Nested
+        end
+        ]
+
+      assert {:ok, {:module, Example}} = resolve(project, code)
+    end
```

Comment:
I like how concise these tests are. Good work on that. I really would prefer having a test block test a single thing though.